src/env.cc

```diff
@@ -985,15 +985,11 @@ void Environment::Exit(int exit_code) {
                     StackTrace::CurrentStackTrace(
                         isolate(), stack_trace_limit(), StackTrace::kDetailed));
   }
-  if (is_main_thread()) {
-    stop_sub_worker_contexts();
-    DisposePlatform();
-    exit(exit_code);
-  } else {
-    worker_context()->Exit(exit_code);
-  }
+  stop_sub_worker_contexts();
```

Comment:
isn't this an un-necessary call for workers?

Reply:
@gireeshpunathil So … no, but thanks for pointing this out.

My line of thinking here was “both the main thread and Workers perform this call anyway, so why not do it at the same time so that behavior is more similar and therefore less prone to bugs”.

While that was not wrong, it introduced a bug that I wouldn’t have noticed if you hadn’t commented; Because this happens without setting `can_call_into_js()` to `false`, the Workers are stopped and their `exit` events would be emitted. We generally don’t want asynchronous resources to emit events during `process.exit()`, so that’s a bug.

The change here meant that this bug would also occur when running nested Workers, but on the main thread, it has been present before. I’ve opened #32546 fixes that (and this PR is rebased on top of it), and marked this PR as blocked until that one lands, because this is semver-minor and the other PR is semver-patch, and our tooling doesn’t like mixing them.

I’ve also added a commit here that makes `Stop(env)` stop Worker threads – that seems like a sensible thing to do, and achieves my original goal of being more consistent between Worker threads and the main thread.